twake/backend/node/src/core/platform/services/tracker/index.ts

```diff
@@ -0,0 +1,118 @@
+import { Consumes, TwakeService, logger } from "../../framework";
+import TrackerAPI from "./provider";
+import { trackedEventBus } from "../../framework/pubsub";
+import {
+  TrackerEventActions,
+  TrackerDataListener,
+  IdentifyObjectType,
+  TrackedEventType,
+} from "./types";
+import Analytics from "analytics-node";
+
+@Consumes([])
+export default class Tracker extends TwakeService<TrackerAPI> implements TrackerAPI {
+  name = "tracker";
+  version = "1";
+  analytics: Analytics;
+
+  async doInit(): Promise<this> {
+    trackedEventBus.subscribe<TrackerDataListener>(TrackerEventActions.TWAKE_OPEN_CLIENT, data => {
+      this.identify({
+        userId: data.user.id,
+      });
+      this.track(
+        {
+          userId: data.user.id,
+          event: TrackerEventActions.TWAKE_OPEN_CLIENT,
+        },
+        (err: Error) => (err ? logger.error("Error while tracking -->", err) : false),
+      );
```

Comment:
The `identity` and `track` functions are async, do you want to call identity then track or both in any order?